include/rmm/device_scalar.hpp

```diff
@@ -37,18 +37,26 @@ class device_scalar {
  public:
   static_assert(std::is_trivially_copyable<T>::value, "Scalar type must be trivially copyable");
 
-  using value_type      = typename device_uvector<T>::value_type;
-  using reference       = typename device_uvector<T>::reference;
-  using const_reference = typename device_uvector<T>::const_reference;
-  using pointer         = typename device_uvector<T>::pointer;
-  using const_pointer   = typename device_uvector<T>::const_pointer;
+  using value_type =
+    typename device_uvector<T>::value_type;  ///< The type of the elements in the scalar
+  using reference       = typename device_uvector<T>::reference;        ///< Not used
+  using const_reference = typename device_uvector<T>::const_reference;  ///< Not used
```

Comment:
Is there a better thing to say here? If it's not used, can it be removed in a follow-up PR?

Reply:
These fields are exposed by `std::vector` so I suspect they are all here for interface compatibility (IOW, downstream consumers might be using them).

Reply:
If you must document everything, then describe what they are, even if they are not used. They absolutely may get used in the future.

Reply:
> (IOW, downstream consumers might be using them)

Correct, I didn't want to remove them because that would constitute a breaking change.

> If you must document everything, then describe what they are, even if they are not used.

@harrism what would you recommend as the docstring? The corresponding `device_uvector` docs are about the return type of `operator []`, but that operator is not defined for `device_scalar`.

Reply:
We could take the approach used by cppreference.com. See for example [`std::vector`](https://en.cppreference.com/w/cpp/container/vector). The approach used there is quite code-centric. E.g.:

| Member type | Definition |
|--------|--------|
| value_type | T |
| size_type | Unsigned integer type (usually std::size_t) |
| difference_type | Signed integer type (usually std::ptrdiff_t) |
| reference | value_type& |
| const_reference | const value_type& |

Unfortunately doxygen won't autogenerate a nice table like cppreference has. (And I think cppreference is a wiki, so they don't have any better technology for generating STL docs).

I will make some suggested changes.

Reply:
I've applied your suggestions, marking this as resolved now.